src/widgets/VariantTimeDistributionPlot.tsx

```diff
@@ -36,59 +53,288 @@ export const VariantTimeDistributionPlot = ({ country, mutations, matchPercentag
-              hovertemplate: '%{y:.2f}%<extra></extra>',
-            },
-          ]}
-          layout={{
-            title: '',
-            xaxis: {
-              title: 'Week',
-              type: 'date',
-              tickvals: distribution.map(d => d.x.firstDayInWeek),
-              tickformat: 'W%-V, %Y',
-              hoverformat: 'Week %-V, %Y (from %d.%m.)',
-            },
-            yaxis: {
-              title: 'Number Sequences',
-            },
-            yaxis2: {
-              title: 'Estimated Percentage',
-              overlaying: 'y',
-              side: 'right',
-            },
-            showlegend: false,
-            margin: { t: 10 },
-          }}
-          config={{
-            displaylogo: false,
-            modeBarButtons: [['zoom2d', 'toImage', 'resetScale2d', 'pan2d']],
-            responsive: true,
-          }}
-        />
+    <MetricWrapper id='metric-with-tooltip'>
+      <div data-for={tooltipId} data-tip={helpText}>
+        <ValueWrapper color={color}>
+          {value}
+          {percent && '%'}
+        </ValueWrapper>
+        <MetricTitleWrapper id='metric-title'>
+          {title + ' '}
+          <IconWrapper id='info-wrapper'>
+            <BiHelpCircle />
+          </IconWrapper>
+        </MetricTitleWrapper>
+      </div>
+      <ReactTooltip id={tooltipId} />
+    </MetricWrapper>
+  );
+};
+
+type CustomTickProps = {
+  x?: number;
+  y?: number;
+  stroke?: unknown;
+  payload?: { value: string };
+  activeIndex: number;
+  dataLength: number;
+  currentValue: string;
+};
+
+const getTickText = (value: string, dataLength: number, isActive: boolean) => {
+  if (dataLength > 10) {
+    return value.slice(5);
+  } else if (dataLength > 5) {
+    return value.slice(2);
+  } else {
+    return value;
+  }
+};
+
+const CustomTick = ({
+  x,
+  y,
+  payload,
+  activeIndex,
+  dataLength,
+  currentValue,
+}: CustomTickProps): JSX.Element => {
+  return (
+    <g transform={`translate(${x},${y})`}>
+      {payload ? (
+        <text
+          x={0}
+          y={0}
+          dx={0}
+          dy={10}
+          textAnchor='middle'
+          fill={payload.value === currentValue ? colors.active : colors.inactive}
+        >
+          {getTickText(payload.value, dataLength, payload.value === currentValue)}
+        </text>
+      ) : (
+        <></>
       )}
-    </div>
+    </g>
   );
 };
 
+export type TimeEntry = {
+  firstDayInWeek: string;
+  yearWeek: string;
+  percent: number;
+  quantity: number;
+};
+
+export type TimeGraphProps = {
+  data: TimeEntry[];
+  height?: number;
+  onClickHandler?: OnClickHandler;
+};
+
+export const TimeGraph = React.memo(
+  ({ data, onClickHandler, height = CHART_HEIGHT }: TimeGraphProps): JSX.Element => {
+    const [activeIndex, setActiveIndex] = useState<number>(data.length - 1);
+    const [ready, setReady] = useState(false);
+    const [currentData, setCurrentData] = useState<TimeEntry>(data[data.length - 1]);
```

Comment:
If `data` is empty you do get some warnings from recharts though. `Received NaN for the `y1` attribute. If this is expected, cast the value to a string.` That's fine for now, since they don't crash the application.

You can search for `N:R203K, N:G204R, S:N501Y, S:A123B, S:N507Y, N:R555Y` and with 100% match percentage to test with empty `data`.